### DISCUSSION TITLE
Language for consistent conversion

### DISCUSSION THREAD

**Discussion by sandeep026** [jgm/pandoc#9967] [Q&A]
I want to write documents in language A and convert them and distribute them into other languages B,C,D,E.... Which language should A follow to get consistent and accurate outputs?

**Comment by jgm** [comment]
By 'language' do you mean 'document format' (e.g. HTML) or natural language (e.g. German)?

**Reply by sandeep026** [reply]
Sorry for not being clear. I meant the document format like md,rst,etc..

**Comment by massifrg** [comment]
`native` and `json` are the ones carrying the maximum of information you may have in a Pandoc document.

As far as I know, every other format handled by pandoc is somehow lossy.

Unfortunately native and json are both hard to read by a person, or edited directly with a text editor.
These are important aspects, in particular thinking of long-term storage.

I'd add two more aspects of a document format:

- being indexable, by Lucene-derived softwares like elasticsearch, docfetcher or solr;

- maintainability with git or any other versioning system.

So we have 5 aspects:

1. quantity of information, especially semantic content

2. human readability

3. editability, especially with a text editor

4. indexability

5. maintainability with versioning systems

Markdown is pretty complete and good considering all those aspects.

Djot is even better, since it adds more consistency and machine efficiency (and being jgm its author, you can expect long-term attention on compatibility with Pandoc).

Complex tables, figures and citations are the most important feature where markdown and djot -- currently -- can't pair Pandoc's native/json richness. See jgm reply in [this discussion](https://github.com/jgm/djot/discussions/278).

Sometimes I fancy of an XML equivalent of pandoc's native format: same content, just a different form.
A Pandoc custom writer for that  should not be that hard.
A custom reader is a bit more complex, because it needs XML parsing, but it's doable.

That would be good enough for the first 4 aspects, while opening to XML tools.
Unfortunately, XML does not work well with diff and merge tools that are common in the git space.

**Reply by sandeep026** [reply]
I mainly make notes for academic context, for which `md` and `rst` are good for taking quick notes. Latex gives finer control but it is more verbose. Among `md` and `rst`, which would be suggested?

**Reply by massifrg** [reply]
I don't know `rst` enough and I don't work with academic documents, so I can't give you well informed advice on `md` vs `rst` (thank you anyway for inducing me to look deeper at their differences).

I just searched for "markdown vs restructuredtext" and there is some good advice available.

I found [this](https://www.dewanahmed.com/markdown-asciidoc-restructuredtext/) interesting, because it makes the comparison in the perspective of maintaining docs as you do with code.
